admin/urls.go

```diff
@@ -320,6 +320,18 @@ func (u *URLs) ReportEdit(org, project, report string) string {
 	return urlutil.MustJoinURL(u.Frontend(), org, project, "-", "reports", report)
 }
 
+// ReportOpenExternal returns the URL for opening a report in the frontend for an external user.
+func (u *URLs) ReportOpenExternal(org, project, report, token string, executionTime time.Time) string {
+	reportURL := urlutil.MustJoinURL(u.Frontend(), org, project, "-", "reports", report, "open", token)
+	reportURL += fmt.Sprintf("?execution_time=%s", executionTime.UTC().Format(time.RFC3339))
+	return reportURL
+}
+
+// ReportExportExternal returns the URL for exporting a report in the frontend for an external user.
+func (u *URLs) ReportExportExternal(org, project, report, token string) string {
+	return urlutil.MustJoinURL(u.Frontend(), org, project, "-", "reports", report, "export", token)
+}
```

Comment:
1. Can these be merged into `ReportOpen` and `ReportExport` with `token string` being an optional parameter? Many of the other URL functions here have optional parameters, but currently they try to have one function per route.
2. Did you check with the applications folks if they prefer `<path>/<token>` vs. `<path>/tokens/<token>` vs. `<path>?token=<token>`? I think the latter might be preferable since it's the same route, but not sure.

Reply:
@ericpgreen2 any thoughts on 2 above. I used `<path>/<token>` to be similar to public urls, but whatever is preferable I can make the change

Reply:
have changed to `<path>?token=<token>` for now

Reply:
Yeah, `<path>?token=<token>` is preferable, thanks